### PR TITLE
[MISC] prepare span header for gcc10

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

File: include/seqan3/std/span [hunk 1]
```diff
@@ -19,7 +19,9 @@
 
 #if __has_include(<span>)
 #include <span>
+#endif
 
+#if defined(__cpp_lib_span)
 #include <seqan3/core/platform.hpp>
 #else // __has_include(<span>)
```
Comment (joergi-w) [inline include/seqan3/std/span]: now you should adapt this comment... 💅

Comment (marehr) [reply include/seqan3/std/span]: true xD

File: include/seqan3/std/span [hunk 2]
```diff
@@ -19,9 +19,11 @@
 
 #if __has_include(<span>)
 #include <span>
+#endif
 
+#if defined(__cpp_lib_span)
 #include <seqan3/core/platform.hpp>
-#else // __has_include(<span>)
+#else // defined(__cpp_lib_span)
```
Comment (rrahn) [inline include/seqan3/std/span]: I guess there must be a comment for the closing macro that has still __has_include?

File: include/seqan3/std/span [hunk 3]
```diff
@@ -19,9 +19,11 @@
 
 #if __has_include(<span>)
 #include <span>
+#endif
```
Comment (rrahn) [inline include/seqan3/std/span]: what if __has_include(<span>) is true but cpp_lib_span isn't? Can we guarantee that this never happens?

Comment (marehr) [reply include/seqan3/std/span]: The header span is always available >= gcc10. cpp_lib_span will only be defined if you build in c++2a mode.

**Pull request merged**